### PR TITLE
TF-TRT MNIST test with V2 converter

### PR DESCRIPTION
**Pull request by tfeher**
This PR extends the MNIST test to use the V2 converter. We create a saved model, and convert it both with and without dynamic shape mode.

Tagging @bixia1 for review. Tracker #45481

### CODE REVIEW

**Comment by tfeher** [comment]
@bixia1 this is ready for review

File: tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py [hunk 1]
```diff
@@ -43,16 +46,40 @@
...
 from tensorflow.python.training.adam import AdamOptimizer
 from tensorflow.python.training.checkpoint_management import latest_checkpoint
 from tensorflow.python.training.training_util import get_global_step
+from tensorflow.compat.v1 import Session
 
 INPUT_NODE_NAME = 'input'
 OUTPUT_NODE_NAME = 'output'
 
 
+def _PreprocessFn(entry):
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py:66]: Can you document the reason for applying this step for the data set?

File: tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py [hunk 2]
```diff
@@ -281,5 +299,110 @@ def testEval(self):
...
+        func = converter._converted_func
+      else:
+        saved_model_loaded = saved_model_load(
+            saved_model_dir, tags=[tag_constants.SERVING])
+        func = saved_model_loaded.signatures[
+            signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY]
+    return func
+
+  def _Run(self, use_trt, batch_size, model_dir, use_dynamic_shape=False):
+    """Train or evaluate the model.
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py:362]: Does this routine only "evaluate" but not "train" the model?
Looks like we don't follow the rule "using third person singular verb forms in subroutine document" in this whole file.

File: tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py [hunk 3]
```diff
@@ -281,5 +299,110 @@ def testEval(self):
...
+    for example in ds:
+      image, label = example[0], example[1]
+      pred = func(image)
+      m.update_state(math_ops.argmax(pred['output'], axis=1), label)
+
+    return m.result().numpy()
+
+  def testEval(self):
+    model_dir = test.test_src_dir_path(
+      'python/compiler/tensorrt/test/testdata/mnist')
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py:389]: Can we name this constant and use the name for both classes in this file?

File: tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py [hunk 4]
```diff
@@ -281,5 +299,110 @@ def testEval(self):
...
+      The Estimator evaluation result.
+    """
+    func = self._GetFunc(use_trt, model_dir, use_dynamic_shape)
+    ds = _GetDataSet(batch_size)
+
+    m = Accuracy()
+    for example in ds:
+      image, label = example[0], example[1]
+      pred = func(image)
+      m.update_state(math_ops.argmax(pred['output'], axis=1), label)
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py:383]: This method looks so different from the method used in the run method of the first class.
Is the method used in the first class more complicate than necessary?

File: tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py (hunk 3 above)
Comment (tfeher) [reply tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py]: Yes, fixed.

**Review by tfeher — COMMENTED** [review]
Thanks @bixia1 for the comments, I have updated the PR.

**Review by bixia1 — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Comment by gbaned** [comment]
@tfeher Can you please resolve conflicts? Thanks!

**Comment by tfeher** [comment]
@gbaned I have resolved the conflicts.

**Review by bixia1 — APPROVED** [review]

File: tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py [hunk 5]
```diff
@@ -39,14 +42,42 @@
...
+from tensorflow.python.saved_model import tag_constants
+from tensorflow.python.saved_model import utils as saved_model_utils
+from tensorflow.python.saved_model import signature_def_utils
+from tensorflow.python.saved_model.load import load as saved_model_load
 from tensorflow.python.summary import summary
 from tensorflow.python.training import saver
 from tensorflow.python.training.adam import AdamOptimizer
 from tensorflow.python.training.checkpoint_management import latest_checkpoint
 from tensorflow.python.training.training_util import get_global_step
+from tensorflow.compat.v1 import Session
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py:56]: This line doesn't work, you need to replace it with 
from tensorflow.python.client import session

then replace "Session(..)" with "session.Session(...)"

see here for more details
https://github.com/tensorflow/tensorflow/blob/8b202f08d52e8206af2bdb2112a62fafbc546ec7/tensorflow/python/client/session_test.py#L31

**Review by bixia1 — CHANGES_REQUESTED** [review]

**Review by tfeher — COMMENTED** [review]
Fixed the import line.

File: tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py (hunk 5 above)
Comment (tfeher) [reply tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py]: Fixed.

**Review by bixia1 — APPROVED** [review]

**Pull request merged**